platforms/platforms_test.go

```diff
@@ -0,0 +1,137 @@
+package platforms
+
+import (
+	"reflect"
+	"runtime"
+	"testing"
+
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+func TestParseSelector(t *testing.T) {
+	var (
+		defaultOS   = runtime.GOOS
+		defaultArch = runtime.GOARCH
+	)
+
+	for _, testcase := range []struct {
+		input     string
+		expected  specs.Platform
+		formatted string
+	}{
+		{
+			input: "*",
+			expected: specs.Platform{
+				OS:           "*",
+				Architecture: "*",
+			},
+			formatted: "*/*",
+		},
+		{
+			input: "linux/*",
+			expected: specs.Platform{
+				OS:           "linux",
+				Architecture: "*",
+			},
+			formatted: "linux/*",
+		},
+		{
+			input: "*/arm64",
+			expected: specs.Platform{
+				OS:           "*",
+				Architecture: "arm64",
+			},
+			formatted: "*/arm64",
+		},
+		{
+			input: "linux/arm/v6",
+			expected: specs.Platform{
+				OS:           "linux",
+				Architecture: "arm",
+			},
+			formatted: "linux/arm",
```

Comment:
This seems bizarre to me -- is the intended usage of this PR for something like a `--platform` flag to `docker pull` and friends?

I'd personally be very surprised to find that `docker pull --platform linux/arm/v6 xxx` ended up getting converted to simply `linux/arm`, and likewise, I'd be surprised if `--platform linux/arm` implied _only_ `v6`.  I'd honestly want `linux/arm` to throw an explicit specificity error, or to imply `linux/arm/*` (and thus choose the first image which is compatible with my current platform).

Reply:
In Debian armhf is v7 with hardfloat see [this](https://wiki.debian.org/ArmHardFloatPort#Supported_devices) (this is the reason for the existence of Rasbian which rebuilds as v6 for compat with RPi hardware). I believe at the time there was cross distro agreement on that as a baseline (at least Ubuntu and Fedora IIRC) for the ARM distros of the day.

Reply:
Yeah, I think this default might need to change based on where it's being used -- for ARM-based images that don't specify a variant, I think `v7` would be the safer choice/assumption, but for a user pulling images, they probably want to allow `v6` if all they specify is "ARM".

Reply:
@ijc @tianon Fair enough. I'll see if I can work out that detail correctly.

Reply:
@tianon @ijc PTAL